gcc/d/d-codegen.cc

```diff
@@ -607,6 +607,11 @@ build_address (tree exp)
   if (TREE_CODE (exp) == CONST_DECL)
     exp = DECL_INITIAL (exp);
 
+  /* Some expression lowering may request an address of a compile-time constant.
+     Make sure it is assigned to a location we can reference.  */
+  if (CONSTANT_CLASS_P (exp) && TREE_CODE (exp) != STRING_CST)
+    exp = build_target_expr (exp);
+
```

Comment:
I guess this change could be valid for older GCC versions as well?

Reply:
Yes that is correct. The same is also done for constructors, however handled during gimplify.

https://github.com/D-Programming-GDC/GDC/blob/d0614a94c7ef4555181e84b5a14c9aad1cfaa303/gcc/d/d-lang.cc#L847